Sources/NostrSDK/NIP05Validating.swift

```diff
@@ -0,0 +1,157 @@
+//
+//  NIP05Validating.swift
+//  
+//
+//  Created by Bryan Montz on 6/12/23.
+//
+
+import Foundation
+
+/// A constant that describes a type of error encountered while validating a [NIP-05](https://github.com/nostr-protocol/nips/blob/master/05.md#nip-05) identifier.
+public enum NIP05ValidationError: Error {
+    /// The identifier was not in the expected format. Valid identifiers look like email addresses, such as "bob@example.com".
+    case invalidIdentifierFormat
+    
+    /// The URL composed from the local-part and domain of a provided identifier was invalid.
+    case invalidComposedURL
+    
+    /// The pubkey provided did not match the one received from the host.
+    case failedValidation
+}
+
+/// A response to a [NIP-05](https://github.com/nostr-protocol/nips/blob/master/05.md#nip-05) request.
+///
+/// It be decoded from json data that looks like this:
+/// ```json
+/// {
+///   "names": {
+///     "bob": "b0635d6a9851d3aed0cd6c495b282167acf761729078d975fc341b22650b07b9"
+///   },
+///   "relays": {
+///     "b0635d6a9851d3aed0cd6c495b282167acf761729078d975fc341b22650b07b9": [ "wss://relay.example.com", "wss://relay2.example.com" ]
+///   }
+/// }
+/// ```
+struct NIP05Response: Codable {
+    
+    /// An object with names as properties and hex-encoded pubkeys as values.
+    ///
+    /// Use this to look up the pubkey associated with an email-like [internet identifier](https://datatracker.ietf.org/doc/html/rfc5322#section-3.4.1).
+    let names: [String: String]?
+    
+    /// An object with public keys as properties and arrays of relay URLs as values
+    ///
+    /// Use this to learn in which relays a specific user may be found.
+    let relays: [String: [String]]?
+}
+
+public protocol DataRequesting {
```

Comment:
Should we refactor this and other generic HTTP request response stuff out? I imagine there could be other NIPs that might use it.